mobile/app/sign-up.tsx

```diff
@@ -40,11 +53,51 @@ export default function Page() {
     return unsubscribe;
   }, [navigation]);
 
+  useEffect(() => {
+    (async () => {
+      console.log("signUpState ->", signUpState);
+
+      if (signUpState === SignUpState.user_exists_on_blockchain_and_local_storage) {
+        setError(
+          "This name is already registered on the blockchain and on this device. Please choose another name or press Back for a normal sign in."
+        );
+        return;
+      }
+      if (signUpState === SignUpState.user_already_exists_on_blockchain) {
+        setError("This name is already registered on the blockchain. Please, choose another name.");
+        return;
+      }
+      if (signUpState === SignUpState.user_already_exists_on_blockchain_under_different_name) {
+        setError(
+          "This account is already registered on the blockchain under different name. Please choose another Seed Phrase or press Back for a normal sign in."
```

Comment:
The logic for this use case is correct. The error message "Please choose another Seed Phrase" may be confusing because there is no button to choose another seed phrase. Also "press Back for a normal sign in" may be confusing if there is no existing account to sign it with. I would suggest changing the message to "This account is already registered on the blockchain under different name. Please press Back and sign up again with another Seed Phrase, or for a normal sign in with a different account if available."